### ISSUE TITLE
Gmail GSuite Email Login Deprecated

### ISSUE THREAD

**Issue by deuks** [salesagility/SuiteCRM#8491]
Gmail will no longer be usable with the current smtp implementation of SuiteCRM in 2021.

My company currently uses GSuite to send/manage email.  Currently we use our login information in SuiteCRM in order to send emails from there.  This involves turning on "Allow Less Secure Apps" on each Google Account that is sending/receiving mail to SuiteCRM.

As of February 15th, 2021, Google is removing the ability to use the "Allow Less Secure Apps" and is instead moving to some sort of OAuth system involving keys, etc. When that day comes, the current style of using gmail credentials to login via SuiteCRM will no longer be supported.

I saw someone bring this up in the SuiteCRM Forums over a month ago, but I don't think they posted here like PGR asked.

**Comment by Mac-Rae** [comment]
Hi @deuks,

This feature is currently under development and will be in place well in advance of the deadline :+1:

**Comment by ItzikHarel** [comment]
Ahoy there
Turns out new comers to SuiteCRM using Google Gsuite (like myself) are already blocked out as first time connection using the "old" protocol are blocked starting June 15, 2020.
Reference: https://github.com/freescout-helpdesk/freescout/issues/390
Is there any ETA on this item? can this change be made available for beta testing?

Thanks, Itzik.

**Comment by pstevens71** [comment]
Has anyone tried Google App Passwords as a work around?
[https://support.google.com/accounts/answer/185833?hl=en](url)

**Comment by pstevens71** [comment]
I just got around to trying it myself.  I don't have Gsuite, but App Passwords do in fact work with normal Gmail.  I can confirm this works.  If anyone has Gsuite, could they try it out, I'd love to know for sure.  I have clients that use Gsuite.

**Comment by deuks** [comment]
This should work as a workaround, however this does require that 2-Factor is turned on, which may or may not be easily implemented depending on someone's situation.

Only other flaw I could possibly see in this workaround is that email address passwords with special characters are malformed and do not work in SuiteCRM (this is detailed in #8719 ). From my tests, it looks like Google only uses alphabetical characters for App Passwords, but if this changes in the future then certain App Passwords might not work.

**Comment by pgorod** [comment]
Cancelled!

https://workspaceupdates.googleblog.com/2019/12/less-secure-apps-oauth-google-username-password-incorrect.html

**Comment by marcoblancas** [comment]
Hi... my GSuite account from Education Workspace today display this warning:

> On 30 May 2022, this setting will no longer be available.

![imagen](https://user-images.githubusercontent.com/5545390/158415339-538f0d5f-e261-4507-84b5-f717dba0d670.png)

And link here https://support.google.com/accounts/answer/6010255?hl=en-GB 

If that's not cancelled by Google... in a month some alternative should be deployed.

**Comment by deuks** [comment]
@marcoblancas My workaround for my company has simply been to enable 2-Factor and then create an app password for suitecrm SMTP. 2-Factor is more secure in the first place, app passwords by google are not being removed on the 30th, and app passwords are only alphabetical, so the special character bug for suitecrm won't be an issue.

**Comment by pstevens71** [comment]
hey @pgorod it pretty much just generates a username and password in Google that you use as a regular username and password in SuiteCRM.  At least that's how it worked last time I used it.

**Comment by deuks** [comment]
I agree with pstevens, but my app passwords haven't made me a username. It just generates the app password (write it down because once you close the pop up window you can never view it again, you'll have to delete it and generate a new one), and I use the app password and my normal email address.

**Comment by marcoblancas** [comment]
@deuks, thank you. I have tried that and seems to be working disable the "Less secure app access" and I think that's a good workaround.

I have configured my Outgoing Mail Configuration SuiteCRM with this system.

I'm agree that way is much more secure but enable the two factor authentication for all my users will be painful. **Users are used to authorize third party apps with the tipicall OAuth2 pop-up and something like this should be enabled in SuiteCRM**. I suppose that development is working in this way. 

By other hand, I have been talking with Google and they say that's not going to impact in IMAP access... but you know It was just a chat with the support team and they haven't an Offial Note about how is going to impact this with IMAP. As workaround they point to this workaround related with App Paswords.

I confirm app passwords generate just one password (16 letter) and no a new username account and is recommended to annotate it if you don't want to generate a new password in the future. Users can remove this password in their Google Account config in the same way they remove OAuth2 authorizations for third party apps but they can't recover the generated passwords.

IMO, the workaround https://support.google.com/accounts/answer/185833?hl=en is valid and usable at the moment.
